src/main/java/com/aws/greengrass/config/ConfigurationReader.java

```diff
@@ -110,24 +111,32 @@ private static void mergeTLogInto(Configuration c, Path p) throws IOException {
      */
     public static void validateTlog(Path tlogPath) throws IOException {
         try (BufferedReader in = Files.newBufferedReader(tlogPath)) {
+            // We have seen two different file corruption scenarios
+            // 1. The last line of config file is corrupted with non-UTF8 characters and BufferedReader::readLine
+            // throws a MalformedInputException.
+            // 2. The config file is filled with kilobytes of null bytes and the first line read from file is not
+            // parseable.
+            // To handle both scenarios and make sure we can fall back to backup config files, we decided to validate
+            // the entire file.
+
+            // Specific description of scenario 2:
             // We have been seeing that very rarely the transaction log gets corrupted when a device (specifically
             // raspberry pi using an SD card) has a power outage.
             // The corruption is happening at the hardware level and there really isn't anything that we can do
             // about it right now.
             // The corruption that we see is that the tlog file is filled with kilobytes of null
             // bytes, depending on how large the configuration was before dumping the entire config to disk.
 
-            // When parsing the tlog using a buffered reader, the corrupt section won't be parsable and so we will
-            // throw an exception here. This validate method is specifically targeting this type of corruption where
-            // the first line is corrupted.
-            // The other opportunity for corruption is when we write to the end of the
-            // file and we do not want to throw on that corruption because our reader is already setup to skip over that
-            // type of problem.
-
             String l = in.readLine();
-            // null if EOF
-            if (l != null) {
+            // if file is empty, throw an IOException so that nucleus can recover with backup config files
+            if (l == null) {
+                throw new IOException(String.format("Empty transaction log file at %s", tlogPath));
```

Comment:
I don't think this is appropriate. 

Readline will return null at the end of file which isn't a problem.